### PR TITLE
Fix tooltip memory leak: destroy doesn't destroy $tip & $arrow

### PR DESCRIPTION
**Pull request by nsteenbeek**

### CODE REVIEW

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 9821699d90c91dcecaa4d62d4d32c12f30acfc49
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/53813088

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
@hnrch02 LGTY? Seems okay to me.

**Comment by afinne** [comment]
The Tooltip still seems to leak a bit. What we did in our project was to also clear out the $element. This, however required also an extra check in the complete function of the hide function:

    function complete() {
      if (that.hoverState != 'in') $tip.detach()
      if (that.$element) {
        that.$element
          .removeAttr('aria-describedby')
          .trigger('hidden.bs.' + that.type)
      }
      callback && callback()
    }

And to be extra sure that we are not calling destroy too many times:

    Tooltip.prototype.destroy = function () {
      var that = this
      clearTimeout(this.timeout)
      if (this.isDestroying == true) {
        return
      }
      this.isDestroying = true
      this.hide(function () {
        that.$element.off('.' + that.type).removeData('bs.' + that.type)
        if (that.$tip) {
          that.$tip.detach()
        }
        that.$tip = null
        that.$arrow = null
        that.$viewport = null
        that.$element = null
      })

**Comment by cvrebert** [comment]
@afinne Pleaae open a new pull request or issue.
